ui/packages/catalog-ui-search/src/main/webapp/component/search-form-editor/search-form-editor.view.js

```diff
@@ -95,19 +94,17 @@ module.exports = Marionette.LayoutView.extend({
       return
     }
 
-    let collection
+    const collection = SearchFormCollection.getCollection()
 
     if (id === 'create') {
-      collection = SearchFormsCollection.getCollection()
       this.model = new QueryModel.Model()
     } else {
-      collection = SearchFormsCollection.getCollection()
       this.model = collection.get(id)
-      if (!this.model) {
-        collection = SearchFormsSharingCollection.getCollection()
-        this.model = collection.get(id)
+      if (this.model) {
```

Comment:
When would you not have this?

Reply:
this is a result of how this component listens to the router for a change. If the user goes to create a new search form, then saves it, and then goes to a workspace. The workspace id will get passed into here and the component will crash. Which causes some other bugs in Intrigue. There's probably a better way to ensure the workspace ids don't get passed here but this is the quickest fix.